lesson25/app.js

```diff
@@ -0,0 +1,118 @@
+const checkBox = document.getElementById('js-check');
+const modal = document.getElementById('modal');
+const modalContent = document.getElementById('js-modalContent');
+
+const modalButton = document.getElementById('js-modalButton');
+const closeButton = document.getElementById('js-closeButton');
+const agreeButton = document.getElementById('js-agreeButton');
+const submitButton = document.getElementById('js-submitButton');
+submitButton.disabled = true;
+
+const p = document.createElement("p");
+let errorMsg = "";
+
+const flugs = {
+    name: false,
+    email: false,
+    check: false,
+    password: false
+};
+
+
+function openModal() {
+    modal.style.display = 'block';
+}
+function closeModal() {
+    modal.style.display = 'none';
+}
+
+
+modalButton.addEventListener('click', openModal);
+closeButton.addEventListener('click', closeModal);
+
+
+modalContent.onscroll = function () {
+    const offsetHeight = this.offsetHeight;
+    const scrollHeight = this.scrollHeight;
+    const scrollTop = this.scrollTop;
+    const scrollPosition = offsetHeight + scrollTop;
+    const proximity = 0;
+
+    if ((scrollHeight - scrollPosition) / scrollHeight <= proximity) {
+        agreeButton.disabled = false;
+        agreeButton.addEventListener('click', () => {
+            checkBox.disabled = false;
+            checkBox.checked = true;
+            closeModal();
+            flugs.check = true;
+            checkFlags();
+        });
+    }
+};
+
+function createErrorMessage() {
+    p.textContent = errorMsg;
+    p.classList.add("error-message");
+}
+
+function clearErrorMessage() {
+    errorMsg = "";
+    p.textContent = errorMsg;
+}
+
+
+function handleNameChanges() {
+    const maxCharacters = 16;
+    if (this.value.length >= maxCharacters) {
+        errorMsg = "※ユーザー名は15文字以下にしてください。";
+        createErrorMessage();
+        this.after(p);
+        flugs.name = false;
+    } else {
+        clearErrorMessage();
+        flugs.name = true;
+    }
+    checkFlags();
+}
+
+function handleEMailChanges() {
+    if (!this.value.match(/.+@.+\..+/)) {
```

Comment:
matchだと配列かnullが返ってきますが

```suggestion
    if (/.+@.+\..+/.test(this.value)) {
        // trueのやつ
     }
```
だとbooleanが返ってくるのでより直感的です
正規表現は変数で管理してください

Reply:
修正しました。

99a87e6